File: CultureLinkCRM/CultureLinkCRM.Web/Modules/CrmDB/ContactAddress/ContactAddressColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace CultureLinkCRM.CrmDB.Columns;

[ColumnsScript("CrmDB.ContactAddress")]
[BasedOnRow(typeof(ContactAddressRow), CheckNames = true)]
public class ContactAddressColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ContactAddressId { get; set; }

    [DisplayName("Address"), EditLink]
    public string AddressLine1 { get; set; }

    [DisplayName("City"), EditLink]
    public string CityName { get; set; }

    [DisplayName("State"), EditLink, Width(50)]
    public string JurisdictionName { get; set; }

    [DisplayName("Zip Code"), EditLink, Width(60)]
    public string ZipCodeName { get; set; }

}

File: CultureLinkCRM/CultureLinkCRM.Web/Modules/ServerTypes/CrmDB/ContactAddressColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ContactAddressRow } from "./ContactAddressRow";

export interface ContactAddressColumns {
    ContactAddressId: Column<ContactAddressRow>;
    AddressLine1: Column<ContactAddressRow>;
    CityName: Column<ContactAddressRow>;
    JurisdictionName: Column<ContactAddressRow>;
    ZipCodeName: Column<ContactAddressRow>;
}

export class ContactAddressColumns extends ColumnsBase<ContactAddressRow> {
    static readonly columnsKey = 'CrmDB.ContactAddress';
    static readonly Fields = fieldsProxy<ContactAddressColumns>();
}